pkl-core/src/main/java/org/pkl/core/ast/type/TypeNode.java

```diff
@@ -94,6 +94,14 @@ public Object executeEagerly(VirtualFrame frame, Object value) {
     return execute(frame, value);
   }
 
+  /**
+   * Checks if {@code value} conforms to this type.
+   *
+   * <p>If {@code value} is conforming, sets {@code slot} to {@code value}. Otherwise, throws a
+   * {@link VmTypeMismatchException}.
+   */
+  public abstract Object executeEagerlyAndSet(VirtualFrame frame, Object value);
```

Comment:
This method is used by a function's parameter type node.

This node:
```
function foo(bar: Listing<Int>) =
                  ^^^^^^^^^^^^
```

We need to add the method because we need to capture the logic of: 
1. Run an eager typecheck
2. Set that value on the frame.